packages/js/product-editor/src/hooks/use-conditional-execution.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { RefObject } from 'react';
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useVisibilityObserver } from './use-visibility-observer';
+
+type useConditionalExecutionProps = {
```

Comment:
I think types must be capitalized and I rather prefer its own file for them, but this last point is up to you.

Reply:
Done in [02d2cc0](https://github.com/woocommerce/woocommerce/pull/47025/commits/02d2cc0e8683312b4aa54af3b3d7993097e76873)